{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>127399_SOFT_TISSUE</th>\n",
       "      <th>1321N1_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>2313287_STOMACH</th>\n",
       "      <th>697_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>769P_KIDNEY</th>\n",
       "      <th>786O_KIDNEY</th>\n",
       "      <th>8305C_THYROID</th>\n",
       "      <th>8505C_THYROID</th>\n",
       "      <th>921_UVEA</th>\n",
       "      <th>...</th>\n",
       "      <th>UMUC3_URINARY_TRACT</th>\n",
       "      <th>VMRCRCW_KIDNEY</th>\n",
       "      <th>WM115_SKIN</th>\n",
       "      <th>WM1799_SKIN</th>\n",
       "      <th>WM2664_SKIN</th>\n",
       "      <th>WM793_SKIN</th>\n",
       "      <th>WM88_SKIN</th>\n",
       "      <th>WSUDLCL2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>YD38_UPPER_AERODIGESTIVE_TRACT</th>\n",
       "      <th>YMB1_BREAST</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CDH2 (1000)</td>\n",
       "      <td>-0.089352</td>\n",
       "      <td>-0.002663</td>\n",
       "      <td>-0.137925</td>\n",
       "      <td>0.173165</td>\n",
       "      <td>-0.134545</td>\n",
       "      <td>-0.148781</td>\n",
       "      <td>-0.203150</td>\n",
       "      <td>-0.037625</td>\n",
       "      <td>-0.175778</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.066021</td>\n",
       "      <td>0.107229</td>\n",
       "      <td>-0.227427</td>\n",
       "      <td>-0.032107</td>\n",
       "      <td>-0.218693</td>\n",
       "      <td>-0.013857</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.230569</td>\n",
       "      <td>0.021516</td>\n",
       "      <td>-0.106172</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AKT3 (10000)</td>\n",
       "      <td>-0.131082</td>\n",
       "      <td>0.065231</td>\n",
       "      <td>0.087221</td>\n",
       "      <td>-0.411486</td>\n",
       "      <td>0.177895</td>\n",
       "      <td>0.324306</td>\n",
       "      <td>0.040381</td>\n",
       "      <td>0.022989</td>\n",
       "      <td>-0.360923</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.035967</td>\n",
       "      <td>-0.031698</td>\n",
       "      <td>0.221841</td>\n",
       "      <td>0.207119</td>\n",
       "      <td>-0.175644</td>\n",
       "      <td>-0.063389</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.519001</td>\n",
       "      <td>-0.029192</td>\n",
       "      <td>-0.097869</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>POU5F1P5 (100009667)</td>\n",
       "      <td>-0.212644</td>\n",
       "      <td>-0.547079</td>\n",
       "      <td>-0.159532</td>\n",
       "      <td>-0.277817</td>\n",
       "      <td>0.179144</td>\n",
       "      <td>0.084569</td>\n",
       "      <td>-0.179724</td>\n",
       "      <td>0.000420</td>\n",
       "      <td>-0.091683</td>\n",
       "      <td>...</td>\n",
       "      <td>0.106033</td>\n",
       "      <td>0.060357</td>\n",
       "      <td>0.137065</td>\n",
       "      <td>0.072449</td>\n",
       "      <td>0.274616</td>\n",
       "      <td>0.032009</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.080685</td>\n",
       "      <td>-0.042488</td>\n",
       "      <td>-0.334812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MED6 (10001)</td>\n",
       "      <td>-0.225475</td>\n",
       "      <td>-0.477512</td>\n",
       "      <td>-0.314678</td>\n",
       "      <td>-0.192239</td>\n",
       "      <td>-0.359350</td>\n",
       "      <td>-0.705736</td>\n",
       "      <td>-0.346217</td>\n",
       "      <td>-0.355164</td>\n",
       "      <td>-0.510637</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.249799</td>\n",
       "      <td>-0.773315</td>\n",
       "      <td>-0.403044</td>\n",
       "      <td>-0.241388</td>\n",
       "      <td>-0.154820</td>\n",
       "      <td>-0.564204</td>\n",
       "      <td>-0.267406</td>\n",
       "      <td>-0.231379</td>\n",
       "      <td>-0.775019</td>\n",
       "      <td>-0.254618</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NR2E3 (10002)</td>\n",
       "      <td>-0.096706</td>\n",
       "      <td>-0.123476</td>\n",
       "      <td>-0.138421</td>\n",
       "      <td>-0.873816</td>\n",
       "      <td>-0.043431</td>\n",
       "      <td>-0.200512</td>\n",
       "      <td>-0.181933</td>\n",
       "      <td>-0.192233</td>\n",
       "      <td>0.064916</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.205856</td>\n",
       "      <td>0.024931</td>\n",
       "      <td>0.276181</td>\n",
       "      <td>0.093556</td>\n",
       "      <td>-0.119268</td>\n",
       "      <td>0.046175</td>\n",
       "      <td>0.406302</td>\n",
       "      <td>-0.750619</td>\n",
       "      <td>-0.256513</td>\n",
       "      <td>-0.376421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7970</th>\n",
       "      <td>PPP4R1 (9989)</td>\n",
       "      <td>-0.036766</td>\n",
       "      <td>0.172239</td>\n",
       "      <td>0.138488</td>\n",
       "      <td>-0.212066</td>\n",
       "      <td>0.181223</td>\n",
       "      <td>0.202738</td>\n",
       "      <td>-0.043067</td>\n",
       "      <td>0.082625</td>\n",
       "      <td>0.179550</td>\n",
       "      <td>...</td>\n",
       "      <td>0.012154</td>\n",
       "      <td>0.003735</td>\n",
       "      <td>-0.222169</td>\n",
       "      <td>0.046680</td>\n",
       "      <td>0.031706</td>\n",
       "      <td>0.085820</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.245949</td>\n",
       "      <td>-0.074758</td>\n",
       "      <td>-0.122497</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7971</th>\n",
       "      <td>CDH1 (999)</td>\n",
       "      <td>-0.174443</td>\n",
       "      <td>0.053323</td>\n",
       "      <td>-0.362161</td>\n",
       "      <td>-0.123080</td>\n",
       "      <td>0.007448</td>\n",
       "      <td>0.073178</td>\n",
       "      <td>-0.081611</td>\n",
       "      <td>0.029975</td>\n",
       "      <td>0.147575</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.001828</td>\n",
       "      <td>0.018768</td>\n",
       "      <td>0.064089</td>\n",
       "      <td>0.112538</td>\n",
       "      <td>-0.010692</td>\n",
       "      <td>0.188347</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.077843</td>\n",
       "      <td>-0.038611</td>\n",
       "      <td>-0.033235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7972</th>\n",
       "      <td>PTBP3 (9991)</td>\n",
       "      <td>-0.036799</td>\n",
       "      <td>0.001485</td>\n",
       "      <td>-0.361670</td>\n",
       "      <td>0.187109</td>\n",
       "      <td>-0.291701</td>\n",
       "      <td>-0.440892</td>\n",
       "      <td>-0.082666</td>\n",
       "      <td>-0.346774</td>\n",
       "      <td>-0.544378</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.279045</td>\n",
       "      <td>-0.190559</td>\n",
       "      <td>-1.071656</td>\n",
       "      <td>-0.196719</td>\n",
       "      <td>-0.699364</td>\n",
       "      <td>-0.610545</td>\n",
       "      <td>-1.166492</td>\n",
       "      <td>-0.066923</td>\n",
       "      <td>-0.468342</td>\n",
       "      <td>-0.365915</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7973</th>\n",
       "      <td>KCNE2 (9992)</td>\n",
       "      <td>-0.020674</td>\n",
       "      <td>0.003644</td>\n",
       "      <td>0.002908</td>\n",
       "      <td>-0.080498</td>\n",
       "      <td>0.154795</td>\n",
       "      <td>0.059115</td>\n",
       "      <td>0.178424</td>\n",
       "      <td>0.057289</td>\n",
       "      <td>0.063665</td>\n",
       "      <td>...</td>\n",
       "      <td>0.152772</td>\n",
       "      <td>0.045208</td>\n",
       "      <td>0.131550</td>\n",
       "      <td>0.261826</td>\n",
       "      <td>0.106279</td>\n",
       "      <td>0.006334</td>\n",
       "      <td>-0.059844</td>\n",
       "      <td>-0.088857</td>\n",
       "      <td>-0.004341</td>\n",
       "      <td>-0.031543</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7974</th>\n",
       "      <td>CASP8AP2 (9994)</td>\n",
       "      <td>-0.762784</td>\n",
       "      <td>-0.224090</td>\n",
       "      <td>-0.208999</td>\n",
       "      <td>-0.428950</td>\n",
       "      <td>0.018032</td>\n",
       "      <td>0.016122</td>\n",
       "      <td>-0.898269</td>\n",
       "      <td>-0.550961</td>\n",
       "      <td>-0.580094</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.247897</td>\n",
       "      <td>-0.485356</td>\n",
       "      <td>-0.168850</td>\n",
       "      <td>-0.220189</td>\n",
       "      <td>-0.123039</td>\n",
       "      <td>-0.381054</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.378437</td>\n",
       "      <td>-0.320508</td>\n",
       "      <td>-0.801584</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7975 rows × 398 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                Unnamed: 0  127399_SOFT_TISSUE  1321N1_CENTRAL_NERVOUS_SYSTEM  \\\n",
       "0              CDH2 (1000)           -0.089352                      -0.002663   \n",
       "1             AKT3 (10000)           -0.131082                       0.065231   \n",
       "2     POU5F1P5 (100009667)           -0.212644                      -0.547079   \n",
       "3             MED6 (10001)           -0.225475                      -0.477512   \n",
       "4            NR2E3 (10002)           -0.096706                      -0.123476   \n",
       "...                    ...                 ...                            ...   \n",
       "7970         PPP4R1 (9989)           -0.036766                       0.172239   \n",
       "7971            CDH1 (999)           -0.174443                       0.053323   \n",
       "7972          PTBP3 (9991)           -0.036799                       0.001485   \n",
       "7973          KCNE2 (9992)           -0.020674                       0.003644   \n",
       "7974       CASP8AP2 (9994)           -0.762784                      -0.224090   \n",
       "\n",
       "      2313287_STOMACH  697_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  769P_KIDNEY  \\\n",
       "0           -0.137925                                0.173165    -0.134545   \n",
       "1            0.087221                               -0.411486     0.177895   \n",
       "2           -0.159532                               -0.277817     0.179144   \n",
       "3           -0.314678                               -0.192239    -0.359350   \n",
       "4           -0.138421                               -0.873816    -0.043431   \n",
       "...               ...                                     ...          ...   \n",
       "7970         0.138488                               -0.212066     0.181223   \n",
       "7971        -0.362161                               -0.123080     0.007448   \n",
       "7972        -0.361670                                0.187109    -0.291701   \n",
       "7973         0.002908                               -0.080498     0.154795   \n",
       "7974        -0.208999                               -0.428950     0.018032   \n",
       "\n",
       "      786O_KIDNEY  8305C_THYROID  8505C_THYROID  921_UVEA  ...  \\\n",
       "0       -0.148781      -0.203150      -0.037625 -0.175778  ...   \n",
       "1        0.324306       0.040381       0.022989 -0.360923  ...   \n",
       "2        0.084569      -0.179724       0.000420 -0.091683  ...   \n",
       "3       -0.705736      -0.346217      -0.355164 -0.510637  ...   \n",
       "4       -0.200512      -0.181933      -0.192233  0.064916  ...   \n",
       "...           ...            ...            ...       ...  ...   \n",
       "7970     0.202738      -0.043067       0.082625  0.179550  ...   \n",
       "7971     0.073178      -0.081611       0.029975  0.147575  ...   \n",
       "7972    -0.440892      -0.082666      -0.346774 -0.544378  ...   \n",
       "7973     0.059115       0.178424       0.057289  0.063665  ...   \n",
       "7974     0.016122      -0.898269      -0.550961 -0.580094  ...   \n",
       "\n",
       "      UMUC3_URINARY_TRACT  VMRCRCW_KIDNEY  WM115_SKIN  WM1799_SKIN  \\\n",
       "0               -0.066021        0.107229   -0.227427    -0.032107   \n",
       "1               -0.035967       -0.031698    0.221841     0.207119   \n",
       "2                0.106033        0.060357    0.137065     0.072449   \n",
       "3               -0.249799       -0.773315   -0.403044    -0.241388   \n",
       "4               -0.205856        0.024931    0.276181     0.093556   \n",
       "...                   ...             ...         ...          ...   \n",
       "7970             0.012154        0.003735   -0.222169     0.046680   \n",
       "7971            -0.001828        0.018768    0.064089     0.112538   \n",
       "7972            -0.279045       -0.190559   -1.071656    -0.196719   \n",
       "7973             0.152772        0.045208    0.131550     0.261826   \n",
       "7974            -0.247897       -0.485356   -0.168850    -0.220189   \n",
       "\n",
       "      WM2664_SKIN  WM793_SKIN  WM88_SKIN  \\\n",
       "0       -0.218693   -0.013857        NaN   \n",
       "1       -0.175644   -0.063389        NaN   \n",
       "2        0.274616    0.032009        NaN   \n",
       "3       -0.154820   -0.564204  -0.267406   \n",
       "4       -0.119268    0.046175   0.406302   \n",
       "...           ...         ...        ...   \n",
       "7970     0.031706    0.085820        NaN   \n",
       "7971    -0.010692    0.188347        NaN   \n",
       "7972    -0.699364   -0.610545  -1.166492   \n",
       "7973     0.106279    0.006334  -0.059844   \n",
       "7974    -0.123039   -0.381054        NaN   \n",
       "\n",
       "      WSUDLCL2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \\\n",
       "0                                       -0.230569   \n",
       "1                                        0.519001   \n",
       "2                                       -0.080685   \n",
       "3                                       -0.231379   \n",
       "4                                       -0.750619   \n",
       "...                                           ...   \n",
       "7970                                    -0.245949   \n",
       "7971                                     0.077843   \n",
       "7972                                    -0.066923   \n",
       "7973                                    -0.088857   \n",
       "7974                                    -0.378437   \n",
       "\n",
       "      YD38_UPPER_AERODIGESTIVE_TRACT  YMB1_BREAST  \n",
       "0                           0.021516    -0.106172  \n",
       "1                          -0.029192    -0.097869  \n",
       "2                          -0.042488    -0.334812  \n",
       "3                          -0.775019    -0.254618  \n",
       "4                          -0.256513    -0.376421  \n",
       "...                              ...          ...  \n",
       "7970                       -0.074758    -0.122497  \n",
       "7971                       -0.038611    -0.033235  \n",
       "7972                       -0.468342    -0.365915  \n",
       "7973                       -0.004341    -0.031543  \n",
       "7974                       -0.320508    -0.801584  \n",
       "\n",
       "[7975 rows x 398 columns]"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# following Charlene's tutorial \n",
    "# the goal is to find genes essential to HepG2 (cancer cell line)\n",
    "\n",
    "import pandas as pd\n",
    "from pybedtools import BedTool\n",
    "dep_score = pd.read_csv('D2_DRIVE_gene_dep_scores.csv')\n",
    "dep_score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "# split the names out\n",
    "dep_score['Symbol'] = dep_score['Unnamed: 0'].str.split(' ', expand = True)[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>127399_SOFT_TISSUE</th>\n",
       "      <th>1321N1_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <th>2313287_STOMACH</th>\n",
       "      <th>697_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>769P_KIDNEY</th>\n",
       "      <th>786O_KIDNEY</th>\n",
       "      <th>8305C_THYROID</th>\n",
       "      <th>8505C_THYROID</th>\n",
       "      <th>921_UVEA</th>\n",
       "      <th>A101D_SKIN</th>\n",
       "      <th>...</th>\n",
       "      <th>UMUC3_URINARY_TRACT</th>\n",
       "      <th>VMRCRCW_KIDNEY</th>\n",
       "      <th>WM115_SKIN</th>\n",
       "      <th>WM1799_SKIN</th>\n",
       "      <th>WM2664_SKIN</th>\n",
       "      <th>WM793_SKIN</th>\n",
       "      <th>WM88_SKIN</th>\n",
       "      <th>WSUDLCL2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</th>\n",
       "      <th>YD38_UPPER_AERODIGESTIVE_TRACT</th>\n",
       "      <th>YMB1_BREAST</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Symbol</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CDH2</th>\n",
       "      <td>-0.089352</td>\n",
       "      <td>-0.002663</td>\n",
       "      <td>-0.137925</td>\n",
       "      <td>0.173165</td>\n",
       "      <td>-0.134545</td>\n",
       "      <td>-0.148781</td>\n",
       "      <td>-0.203150</td>\n",
       "      <td>-0.037625</td>\n",
       "      <td>-0.175778</td>\n",
       "      <td>0.139467</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.066021</td>\n",
       "      <td>0.107229</td>\n",
       "      <td>-0.227427</td>\n",
       "      <td>-0.032107</td>\n",
       "      <td>-0.218693</td>\n",
       "      <td>-0.013857</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.230569</td>\n",
       "      <td>0.021516</td>\n",
       "      <td>-0.106172</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AKT3</th>\n",
       "      <td>-0.131082</td>\n",
       "      <td>0.065231</td>\n",
       "      <td>0.087221</td>\n",
       "      <td>-0.411486</td>\n",
       "      <td>0.177895</td>\n",
       "      <td>0.324306</td>\n",
       "      <td>0.040381</td>\n",
       "      <td>0.022989</td>\n",
       "      <td>-0.360923</td>\n",
       "      <td>-0.094837</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.035967</td>\n",
       "      <td>-0.031698</td>\n",
       "      <td>0.221841</td>\n",
       "      <td>0.207119</td>\n",
       "      <td>-0.175644</td>\n",
       "      <td>-0.063389</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.519001</td>\n",
       "      <td>-0.029192</td>\n",
       "      <td>-0.097869</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>POU5F1P5</th>\n",
       "      <td>-0.212644</td>\n",
       "      <td>-0.547079</td>\n",
       "      <td>-0.159532</td>\n",
       "      <td>-0.277817</td>\n",
       "      <td>0.179144</td>\n",
       "      <td>0.084569</td>\n",
       "      <td>-0.179724</td>\n",
       "      <td>0.000420</td>\n",
       "      <td>-0.091683</td>\n",
       "      <td>0.439111</td>\n",
       "      <td>...</td>\n",
       "      <td>0.106033</td>\n",
       "      <td>0.060357</td>\n",
       "      <td>0.137065</td>\n",
       "      <td>0.072449</td>\n",
       "      <td>0.274616</td>\n",
       "      <td>0.032009</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.080685</td>\n",
       "      <td>-0.042488</td>\n",
       "      <td>-0.334812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MED6</th>\n",
       "      <td>-0.225475</td>\n",
       "      <td>-0.477512</td>\n",
       "      <td>-0.314678</td>\n",
       "      <td>-0.192239</td>\n",
       "      <td>-0.359350</td>\n",
       "      <td>-0.705736</td>\n",
       "      <td>-0.346217</td>\n",
       "      <td>-0.355164</td>\n",
       "      <td>-0.510637</td>\n",
       "      <td>-0.244747</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.249799</td>\n",
       "      <td>-0.773315</td>\n",
       "      <td>-0.403044</td>\n",
       "      <td>-0.241388</td>\n",
       "      <td>-0.154820</td>\n",
       "      <td>-0.564204</td>\n",
       "      <td>-0.267406</td>\n",
       "      <td>-0.231379</td>\n",
       "      <td>-0.775019</td>\n",
       "      <td>-0.254618</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NR2E3</th>\n",
       "      <td>-0.096706</td>\n",
       "      <td>-0.123476</td>\n",
       "      <td>-0.138421</td>\n",
       "      <td>-0.873816</td>\n",
       "      <td>-0.043431</td>\n",
       "      <td>-0.200512</td>\n",
       "      <td>-0.181933</td>\n",
       "      <td>-0.192233</td>\n",
       "      <td>0.064916</td>\n",
       "      <td>0.163630</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.205856</td>\n",
       "      <td>0.024931</td>\n",
       "      <td>0.276181</td>\n",
       "      <td>0.093556</td>\n",
       "      <td>-0.119268</td>\n",
       "      <td>0.046175</td>\n",
       "      <td>0.406302</td>\n",
       "      <td>-0.750619</td>\n",
       "      <td>-0.256513</td>\n",
       "      <td>-0.376421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PPP4R1</th>\n",
       "      <td>-0.036766</td>\n",
       "      <td>0.172239</td>\n",
       "      <td>0.138488</td>\n",
       "      <td>-0.212066</td>\n",
       "      <td>0.181223</td>\n",
       "      <td>0.202738</td>\n",
       "      <td>-0.043067</td>\n",
       "      <td>0.082625</td>\n",
       "      <td>0.179550</td>\n",
       "      <td>-0.044767</td>\n",
       "      <td>...</td>\n",
       "      <td>0.012154</td>\n",
       "      <td>0.003735</td>\n",
       "      <td>-0.222169</td>\n",
       "      <td>0.046680</td>\n",
       "      <td>0.031706</td>\n",
       "      <td>0.085820</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.245949</td>\n",
       "      <td>-0.074758</td>\n",
       "      <td>-0.122497</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CDH1</th>\n",
       "      <td>-0.174443</td>\n",
       "      <td>0.053323</td>\n",
       "      <td>-0.362161</td>\n",
       "      <td>-0.123080</td>\n",
       "      <td>0.007448</td>\n",
       "      <td>0.073178</td>\n",
       "      <td>-0.081611</td>\n",
       "      <td>0.029975</td>\n",
       "      <td>0.147575</td>\n",
       "      <td>0.091671</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.001828</td>\n",
       "      <td>0.018768</td>\n",
       "      <td>0.064089</td>\n",
       "      <td>0.112538</td>\n",
       "      <td>-0.010692</td>\n",
       "      <td>0.188347</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.077843</td>\n",
       "      <td>-0.038611</td>\n",
       "      <td>-0.033235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PTBP3</th>\n",
       "      <td>-0.036799</td>\n",
       "      <td>0.001485</td>\n",
       "      <td>-0.361670</td>\n",
       "      <td>0.187109</td>\n",
       "      <td>-0.291701</td>\n",
       "      <td>-0.440892</td>\n",
       "      <td>-0.082666</td>\n",
       "      <td>-0.346774</td>\n",
       "      <td>-0.544378</td>\n",
       "      <td>-0.417071</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.279045</td>\n",
       "      <td>-0.190559</td>\n",
       "      <td>-1.071656</td>\n",
       "      <td>-0.196719</td>\n",
       "      <td>-0.699364</td>\n",
       "      <td>-0.610545</td>\n",
       "      <td>-1.166492</td>\n",
       "      <td>-0.066923</td>\n",
       "      <td>-0.468342</td>\n",
       "      <td>-0.365915</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KCNE2</th>\n",
       "      <td>-0.020674</td>\n",
       "      <td>0.003644</td>\n",
       "      <td>0.002908</td>\n",
       "      <td>-0.080498</td>\n",
       "      <td>0.154795</td>\n",
       "      <td>0.059115</td>\n",
       "      <td>0.178424</td>\n",
       "      <td>0.057289</td>\n",
       "      <td>0.063665</td>\n",
       "      <td>0.045011</td>\n",
       "      <td>...</td>\n",
       "      <td>0.152772</td>\n",
       "      <td>0.045208</td>\n",
       "      <td>0.131550</td>\n",
       "      <td>0.261826</td>\n",
       "      <td>0.106279</td>\n",
       "      <td>0.006334</td>\n",
       "      <td>-0.059844</td>\n",
       "      <td>-0.088857</td>\n",
       "      <td>-0.004341</td>\n",
       "      <td>-0.031543</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CASP8AP2</th>\n",
       "      <td>-0.762784</td>\n",
       "      <td>-0.224090</td>\n",
       "      <td>-0.208999</td>\n",
       "      <td>-0.428950</td>\n",
       "      <td>0.018032</td>\n",
       "      <td>0.016122</td>\n",
       "      <td>-0.898269</td>\n",
       "      <td>-0.550961</td>\n",
       "      <td>-0.580094</td>\n",
       "      <td>-0.362667</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.247897</td>\n",
       "      <td>-0.485356</td>\n",
       "      <td>-0.168850</td>\n",
       "      <td>-0.220189</td>\n",
       "      <td>-0.123039</td>\n",
       "      <td>-0.381054</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.378437</td>\n",
       "      <td>-0.320508</td>\n",
       "      <td>-0.801584</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7975 rows × 397 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          127399_SOFT_TISSUE  1321N1_CENTRAL_NERVOUS_SYSTEM  2313287_STOMACH  \\\n",
       "Symbol                                                                         \n",
       "CDH2               -0.089352                      -0.002663        -0.137925   \n",
       "AKT3               -0.131082                       0.065231         0.087221   \n",
       "POU5F1P5           -0.212644                      -0.547079        -0.159532   \n",
       "MED6               -0.225475                      -0.477512        -0.314678   \n",
       "NR2E3              -0.096706                      -0.123476        -0.138421   \n",
       "...                      ...                            ...              ...   \n",
       "PPP4R1             -0.036766                       0.172239         0.138488   \n",
       "CDH1               -0.174443                       0.053323        -0.362161   \n",
       "PTBP3              -0.036799                       0.001485        -0.361670   \n",
       "KCNE2              -0.020674                       0.003644         0.002908   \n",
       "CASP8AP2           -0.762784                      -0.224090        -0.208999   \n",
       "\n",
       "          697_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  769P_KIDNEY  786O_KIDNEY  \\\n",
       "Symbol                                                                       \n",
       "CDH2                                    0.173165    -0.134545    -0.148781   \n",
       "AKT3                                   -0.411486     0.177895     0.324306   \n",
       "POU5F1P5                               -0.277817     0.179144     0.084569   \n",
       "MED6                                   -0.192239    -0.359350    -0.705736   \n",
       "NR2E3                                  -0.873816    -0.043431    -0.200512   \n",
       "...                                          ...          ...          ...   \n",
       "PPP4R1                                 -0.212066     0.181223     0.202738   \n",
       "CDH1                                   -0.123080     0.007448     0.073178   \n",
       "PTBP3                                   0.187109    -0.291701    -0.440892   \n",
       "KCNE2                                  -0.080498     0.154795     0.059115   \n",
       "CASP8AP2                               -0.428950     0.018032     0.016122   \n",
       "\n",
       "          8305C_THYROID  8505C_THYROID  921_UVEA  A101D_SKIN  ...  \\\n",
       "Symbol                                                        ...   \n",
       "CDH2          -0.203150      -0.037625 -0.175778    0.139467  ...   \n",
       "AKT3           0.040381       0.022989 -0.360923   -0.094837  ...   \n",
       "POU5F1P5      -0.179724       0.000420 -0.091683    0.439111  ...   \n",
       "MED6          -0.346217      -0.355164 -0.510637   -0.244747  ...   \n",
       "NR2E3         -0.181933      -0.192233  0.064916    0.163630  ...   \n",
       "...                 ...            ...       ...         ...  ...   \n",
       "PPP4R1        -0.043067       0.082625  0.179550   -0.044767  ...   \n",
       "CDH1          -0.081611       0.029975  0.147575    0.091671  ...   \n",
       "PTBP3         -0.082666      -0.346774 -0.544378   -0.417071  ...   \n",
       "KCNE2          0.178424       0.057289  0.063665    0.045011  ...   \n",
       "CASP8AP2      -0.898269      -0.550961 -0.580094   -0.362667  ...   \n",
       "\n",
       "          UMUC3_URINARY_TRACT  VMRCRCW_KIDNEY  WM115_SKIN  WM1799_SKIN  \\\n",
       "Symbol                                                                   \n",
       "CDH2                -0.066021        0.107229   -0.227427    -0.032107   \n",
       "AKT3                -0.035967       -0.031698    0.221841     0.207119   \n",
       "POU5F1P5             0.106033        0.060357    0.137065     0.072449   \n",
       "MED6                -0.249799       -0.773315   -0.403044    -0.241388   \n",
       "NR2E3               -0.205856        0.024931    0.276181     0.093556   \n",
       "...                       ...             ...         ...          ...   \n",
       "PPP4R1               0.012154        0.003735   -0.222169     0.046680   \n",
       "CDH1                -0.001828        0.018768    0.064089     0.112538   \n",
       "PTBP3               -0.279045       -0.190559   -1.071656    -0.196719   \n",
       "KCNE2                0.152772        0.045208    0.131550     0.261826   \n",
       "CASP8AP2            -0.247897       -0.485356   -0.168850    -0.220189   \n",
       "\n",
       "          WM2664_SKIN  WM793_SKIN  WM88_SKIN  \\\n",
       "Symbol                                         \n",
       "CDH2        -0.218693   -0.013857        NaN   \n",
       "AKT3        -0.175644   -0.063389        NaN   \n",
       "POU5F1P5     0.274616    0.032009        NaN   \n",
       "MED6        -0.154820   -0.564204  -0.267406   \n",
       "NR2E3       -0.119268    0.046175   0.406302   \n",
       "...               ...         ...        ...   \n",
       "PPP4R1       0.031706    0.085820        NaN   \n",
       "CDH1        -0.010692    0.188347        NaN   \n",
       "PTBP3       -0.699364   -0.610545  -1.166492   \n",
       "KCNE2        0.106279    0.006334  -0.059844   \n",
       "CASP8AP2    -0.123039   -0.381054        NaN   \n",
       "\n",
       "          WSUDLCL2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \\\n",
       "Symbol                                                  \n",
       "CDH2                                        -0.230569   \n",
       "AKT3                                         0.519001   \n",
       "POU5F1P5                                    -0.080685   \n",
       "MED6                                        -0.231379   \n",
       "NR2E3                                       -0.750619   \n",
       "...                                               ...   \n",
       "PPP4R1                                      -0.245949   \n",
       "CDH1                                         0.077843   \n",
       "PTBP3                                       -0.066923   \n",
       "KCNE2                                       -0.088857   \n",
       "CASP8AP2                                    -0.378437   \n",
       "\n",
       "          YD38_UPPER_AERODIGESTIVE_TRACT  YMB1_BREAST  \n",
       "Symbol                                                 \n",
       "CDH2                            0.021516    -0.106172  \n",
       "AKT3                           -0.029192    -0.097869  \n",
       "POU5F1P5                       -0.042488    -0.334812  \n",
       "MED6                           -0.775019    -0.254618  \n",
       "NR2E3                          -0.256513    -0.376421  \n",
       "...                                  ...          ...  \n",
       "PPP4R1                         -0.074758    -0.122497  \n",
       "CDH1                           -0.038611    -0.033235  \n",
       "PTBP3                          -0.468342    -0.365915  \n",
       "KCNE2                          -0.004341    -0.031543  \n",
       "CASP8AP2                       -0.320508    -0.801584  \n",
       "\n",
       "[7975 rows x 397 columns]"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# drop the unneeded column and make the index Symbol (labels)\n",
    "dep_score.drop('Unnamed: 0', inplace = True, axis = 1)\n",
    "dep_score.set_index('Symbol', inplace = True)\n",
    "dep_score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HEPG2_LIVER</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Symbol</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CDH2</th>\n",
       "      <td>0.057906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AKT3</th>\n",
       "      <td>-0.269006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>POU5F1P5</th>\n",
       "      <td>-0.040908</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MED6</th>\n",
       "      <td>-0.346157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NR2E3</th>\n",
       "      <td>-0.200507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PPP4R1</th>\n",
       "      <td>-0.041508</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CDH1</th>\n",
       "      <td>-0.207714</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PTBP3</th>\n",
       "      <td>-0.188387</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KCNE2</th>\n",
       "      <td>0.041099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CASP8AP2</th>\n",
       "      <td>0.013360</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7975 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          HEPG2_LIVER\n",
       "Symbol               \n",
       "CDH2         0.057906\n",
       "AKT3        -0.269006\n",
       "POU5F1P5    -0.040908\n",
       "MED6        -0.346157\n",
       "NR2E3       -0.200507\n",
       "...               ...\n",
       "PPP4R1      -0.041508\n",
       "CDH1        -0.207714\n",
       "PTBP3       -0.188387\n",
       "KCNE2        0.041099\n",
       "CASP8AP2     0.013360\n",
       "\n",
       "[7975 rows x 1 columns]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get hepg2 targets by making a hepg2 df\n",
    "hepg2 = dep_score[['HEPG2_LIVER']]\n",
    "hepg2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "pandas.core.indexes.base.Index"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "targets = hepg2.loc[hepg2['HEPG2_LIVER'] < -1].index\n",
    "type(targets)\n",
    "# index is a sequence used for sequencing, immutable like string"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['DCAF13P3', 'SMC4', 'ATP6V1G2-DDX39B', 'ALYREF', 'SF3B4',\n",
       "       'U2AF1L5&U2AF1', 'SF3A1', 'RACK1', 'PPIE', 'TACC3',\n",
       "       ...\n",
       "       'CCNK', 'BUD31', 'RPL14', 'PRPF4', 'PRPF3', 'COPB2', 'ZRANB2', 'CKAP5',\n",
       "       'CDK11B', 'RBX1'],\n",
       "      dtype='object', name='Symbol', length=139)"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "targets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "U2AF1L5&U2AF1\n",
      "GPR89B&GPR89A\n"
     ]
    }
   ],
   "source": [
    "# splitting out the & from targets \n",
    "essential_gene = []\n",
    "for t in targets:\n",
    "    if '&' in t:\n",
    "        essential_gene.extend(t.split('&'))\n",
    "        print(t)\n",
    "    else:\n",
    "        essential_gene.append(t)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['DCAF13P3',\n",
       " 'SMC4',\n",
       " 'ATP6V1G2-DDX39B',\n",
       " 'ALYREF',\n",
       " 'SF3B4',\n",
       " 'U2AF1L5',\n",
       " 'U2AF1',\n",
       " 'SF3A1',\n",
       " 'RACK1',\n",
       " 'PPIE',\n",
       " 'TACC3',\n",
       " 'NXF1',\n",
       " 'CCT7',\n",
       " 'CCT2',\n",
       " 'SMC2',\n",
       " 'TBL3',\n",
       " 'USP39',\n",
       " 'RUVBL2',\n",
       " 'WDR3',\n",
       " 'AFG3L2',\n",
       " 'SF3B2',\n",
       " 'RPL35',\n",
       " 'U2AF2',\n",
       " 'AP2M1',\n",
       " 'CLNS1A',\n",
       " 'CLTC',\n",
       " 'NEDD1',\n",
       " 'COPA',\n",
       " 'CTNNB1',\n",
       " 'CYP21A1P',\n",
       " 'DDB1',\n",
       " 'DHX8',\n",
       " 'DHX15',\n",
       " 'DYNC1I2',\n",
       " 'DUSP9',\n",
       " 'EEF2',\n",
       " 'COPZ1',\n",
       " 'PUF60',\n",
       " 'SNRNP200',\n",
       " 'BOP1',\n",
       " 'SF3B1',\n",
       " 'NR5A1',\n",
       " 'XRCC6',\n",
       " 'GAK',\n",
       " 'GEMIN5',\n",
       " 'PRPF31',\n",
       " 'HTATSF1',\n",
       " 'PRPF19',\n",
       " 'NOP53',\n",
       " 'HNRNPC',\n",
       " 'HNRNPU',\n",
       " 'HSPE1',\n",
       " 'IK',\n",
       " 'KIF11',\n",
       " 'SBK1',\n",
       " 'MCM6',\n",
       " 'CRSP8P',\n",
       " 'NACA',\n",
       " 'NRAS',\n",
       " 'ACO2',\n",
       " 'PABPC3',\n",
       " 'DDX47',\n",
       " 'CDC40',\n",
       " 'DDX41',\n",
       " 'GPR89B',\n",
       " 'GPR89A',\n",
       " 'SF3B6',\n",
       " 'ATP6V0C',\n",
       " 'POLA1',\n",
       " 'POLR2A',\n",
       " 'POLR2B',\n",
       " 'POLR2J',\n",
       " 'INO80',\n",
       " 'SMU1',\n",
       " 'PPP6C',\n",
       " 'RBM22',\n",
       " 'PSMA3',\n",
       " 'PSMA4',\n",
       " 'PSMC1',\n",
       " 'PSMC2',\n",
       " 'PSMC3',\n",
       " 'PSMC5',\n",
       " 'PSMD2',\n",
       " 'NAT14',\n",
       " 'RBM25',\n",
       " 'RAN',\n",
       " 'ABCE1',\n",
       " 'RPA1',\n",
       " 'RPA2',\n",
       " 'RPL7',\n",
       " 'RPS16',\n",
       " 'RPS18',\n",
       " 'RPS21',\n",
       " 'RPS27A',\n",
       " 'RRM1',\n",
       " 'S100A1',\n",
       " 'SNRNP70',\n",
       " 'SNRPB',\n",
       " 'SNRPF',\n",
       " 'SSRP1',\n",
       " 'SUPT5H',\n",
       " 'SUPT6H',\n",
       " 'TAF1',\n",
       " 'TCF15',\n",
       " 'TFRC',\n",
       " 'CCT3',\n",
       " 'CDK11A',\n",
       " 'UBA52',\n",
       " 'UBA1',\n",
       " 'VARS',\n",
       " 'VCP',\n",
       " 'XRCC5',\n",
       " 'CDC73',\n",
       " 'SEM1',\n",
       " 'USP5',\n",
       " 'FIP1L1',\n",
       " 'MED25',\n",
       " 'SYMPK',\n",
       " 'TRRAP',\n",
       " 'HIST1H2BO',\n",
       " 'WDR75',\n",
       " 'UTP15',\n",
       " 'PHF5A',\n",
       " 'UTP4',\n",
       " 'ACTL6A',\n",
       " 'RUVBL1',\n",
       " 'EIF3A',\n",
       " 'EIF3B',\n",
       " 'EIF3G',\n",
       " 'EIF3H',\n",
       " 'EIF3I',\n",
       " 'CCNK',\n",
       " 'BUD31',\n",
       " 'RPL14',\n",
       " 'PRPF4',\n",
       " 'PRPF3',\n",
       " 'COPB2',\n",
       " 'ZRANB2',\n",
       " 'CKAP5',\n",
       " 'CDK11B',\n",
       " 'RBX1']"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "essential_gene\n",
    "# these are the transcripts bound\n",
    "# in ENCODE, we have RBPs "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Finding the coordinate for essential genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "# loading transcript annotations\n",
    "transcript_coords = BedTool('/home/hsher/gencode_coords/gencode.v33.transcript.gff3')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "60134"
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(transcript_coords)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\tHAVANA\ttranscript\t11869\t14409\t.\t+\t.\tID=ENST00000456328.2;Parent=ENSG00000223972.5;gene_id=ENSG00000223972.5;transcript_id=ENST00000456328.2;gene_type=transcribed_unprocessed_pseudogene;gene_name=DDX11L1;transcript_type=processed_transcript;transcript_name=DDX11L1-202;level=2;transcript_support_level=1;hgnc_id=HGNC:37102;tag=basic;havana_gene=OTTHUMG00000000961.2;havana_transcript=OTTHUMT00000362751.1\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(transcript_coords[0])\n",
    "# each entry is a transcript\n",
    "# the gene symbol occurs in the transcript_name attribute"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'DDX11L1-202'"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "transcript_coords[0].attrs['transcript_name']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [],
   "source": [
    "# look into bedtools filter and lambda\n",
    "# filtering based on transcript name\n",
    "essential_gene_coords = transcript_coords.filter(lambda t: t.attrs['transcript_name'] in essential_gene).saveas()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "pybedtools.bedtool.BedTool"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# we created new BedTool to hold the coords\n",
    "type(essential_gene_coords)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "querying 1-141...done.\n",
      "Finished.\n",
      "3 input query terms found dup hits:\n",
      "\t[('DCAF13P3', 2), ('ATP6V1G2-DDX39B', 2), ('CYP21A1P', 5)]\n",
      "2 input query terms found no hit:\n",
      "\t['VARS', 'HIST1H2BO']\n",
      "Pass \"returnall=True\" to return complete lists of duplicate or missing query terms.\n"
     ]
    }
   ],
   "source": [
    "# the genes are named differently so need to convert to ensembl transcript ID\n",
    "import mygene\n",
    "mg = mygene.MyGeneInfo()\n",
    "converted = mg.querymany(essential_gene, scopes='symbol',\n",
    "    fields='ensembl.gene,entrezgene', as_dataframe = True, species = 9606)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [],
   "source": [
    "# getting stuck at pandas module must be installed for as_dataframe\n",
    "# the solution was to restart the kernel after installing pandas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>_id</th>\n",
       "      <th>_score</th>\n",
       "      <th>ensembl.gene</th>\n",
       "      <th>entrezgene</th>\n",
       "      <th>ensembl</th>\n",
       "      <th>notfound</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>query</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>DCAF13P3</th>\n",
       "      <td>ENSG00000259378</td>\n",
       "      <td>38.585747</td>\n",
       "      <td>ENSG00000259378</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DCAF13P3</th>\n",
       "      <td>100132724</td>\n",
       "      <td>38.585747</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100132724</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SMC4</th>\n",
       "      <td>10051</td>\n",
       "      <td>86.033905</td>\n",
       "      <td>ENSG00000113810</td>\n",
       "      <td>10051</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATP6V1G2-DDX39B</th>\n",
       "      <td>ENSG00000254870</td>\n",
       "      <td>115.094610</td>\n",
       "      <td>ENSG00000254870</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATP6V1G2-DDX39B</th>\n",
       "      <td>100532737</td>\n",
       "      <td>112.499650</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100532737</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             _id      _score     ensembl.gene entrezgene  \\\n",
       "query                                                                      \n",
       "DCAF13P3         ENSG00000259378   38.585747  ENSG00000259378        NaN   \n",
       "DCAF13P3               100132724   38.585747              NaN  100132724   \n",
       "SMC4                       10051   86.033905  ENSG00000113810      10051   \n",
       "ATP6V1G2-DDX39B  ENSG00000254870  115.094610  ENSG00000254870        NaN   \n",
       "ATP6V1G2-DDX39B        100532737  112.499650              NaN  100532737   \n",
       "\n",
       "                ensembl notfound  \n",
       "query                             \n",
       "DCAF13P3            NaN      NaN  \n",
       "DCAF13P3            NaN      NaN  \n",
       "SMC4                NaN      NaN  \n",
       "ATP6V1G2-DDX39B     NaN      NaN  \n",
       "ATP6V1G2-DDX39B     NaN      NaN  "
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "converted.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "# turning the essential gene ids to a list\n",
    "essential_gene_ids = converted['ensembl.gene'].dropna().tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "# these are both BedTools\n",
    "# FILTER function is useful here\n",
    "essential_gene_coords = transcript_coords.filter(lambda x: x.attrs['Parent'].split('.')[0] in essential_gene_ids).saveas()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "134\n",
      "<class 'list'>\n"
     ]
    }
   ],
   "source": [
    "print(len(essential_gene_ids))\n",
    "print(type(essential_gene_ids))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- found that the length of my essential gene IDs list is different than the reference (mine was 134, reference had 137)\n",
    "- my mygene query also returned different results, but not sure why"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "128"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(essential_gene_coords)\n",
    "# this value is the same as the reference notebook!"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Looking at IDR peaks of essential RBPs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [],
   "source": [
    "# using ENCODE data\n",
    "encode4 = pd.read_csv('/home/hsher/projects/ClipNet/archishma/ENCODE4_1214.csv')\n",
    "encode3 = pd.read_csv('/home/hsher/projects/RBP_annot/ENCODE_FINAL_ANNOTATIONS.uidsonly.txt.manifesthg38.txt', sep = '\\t', header= 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>uid</th>\n",
       "      <th>Batch</th>\n",
       "      <th>RBP</th>\n",
       "      <th>prefix</th>\n",
       "      <th>bam_0</th>\n",
       "      <th>bam_1</th>\n",
       "      <th>bam_control</th>\n",
       "      <th>plus_0</th>\n",
       "      <th>plus_1</th>\n",
       "      <th>plus_control</th>\n",
       "      <th>minus_0</th>\n",
       "      <th>minus_1</th>\n",
       "      <th>minus_control</th>\n",
       "      <th>bed_0</th>\n",
       "      <th>bed_1</th>\n",
       "      <th>idr</th>\n",
       "      <th>Cell Line</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>4001</td>\n",
       "      <td>batch2</td>\n",
       "      <td>NONO</td>\n",
       "      <td>encode4_batch2.4001</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>K562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>4002</td>\n",
       "      <td>batch2</td>\n",
       "      <td>FXR2</td>\n",
       "      <td>encode4_batch2.4002</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>K562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>4004</td>\n",
       "      <td>batch2</td>\n",
       "      <td>DEK</td>\n",
       "      <td>encode4_batch2.4004</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>K562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>4007</td>\n",
       "      <td>batch2</td>\n",
       "      <td>EIF2B3</td>\n",
       "      <td>encode4_batch2.4007</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>K562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>4008</td>\n",
       "      <td>batch2</td>\n",
       "      <td>RBFOX3</td>\n",
       "      <td>encode4_batch2.4008</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>K562</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0   uid   Batch     RBP               prefix  \\\n",
       "0           0  4001  batch2    NONO  encode4_batch2.4001   \n",
       "1           1  4002  batch2    FXR2  encode4_batch2.4002   \n",
       "2           2  4004  batch2     DEK  encode4_batch2.4004   \n",
       "3           3  4007  batch2  EIF2B3  encode4_batch2.4007   \n",
       "4           4  4008  batch2  RBFOX3  encode4_batch2.4008   \n",
       "\n",
       "                                               bam_0  \\\n",
       "0  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "1  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "2  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "3  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "4  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "\n",
       "                                               bam_1  \\\n",
       "0  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "1  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "2  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "3  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "4  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "\n",
       "                                         bam_control  \\\n",
       "0  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "1  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "2  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "3  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "4  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "\n",
       "                                              plus_0  \\\n",
       "0  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "1  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "2  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "3  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "4  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "\n",
       "                                              plus_1  \\\n",
       "0  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "1  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "2  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "3  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "4  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "\n",
       "                                        plus_control  \\\n",
       "0  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "1  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "2  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "3  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "4  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "\n",
       "                                             minus_0  \\\n",
       "0  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "1  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "2  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "3  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "4  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "\n",
       "                                             minus_1  \\\n",
       "0  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "1  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "2  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "3  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "4  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "\n",
       "                                       minus_control  \\\n",
       "0  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "1  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "2  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "3  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "4  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "\n",
       "                                               bed_0  \\\n",
       "0  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "1  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "2  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "3  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "4  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "\n",
       "                                               bed_1  \\\n",
       "0  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "1  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "2  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "3  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "4  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "\n",
       "                                                 idr Cell Line  \n",
       "0  /projects/ps-yeolab5/encore/processing/encore_...      K562  \n",
       "1  /projects/ps-yeolab5/encore/processing/encore_...      K562  \n",
       "2  /projects/ps-yeolab5/encore/processing/encore_...      K562  \n",
       "3  /projects/ps-yeolab5/encore/processing/encore_...      K562  \n",
       "4  /projects/ps-yeolab5/encore/processing/encore_...      K562  "
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "encode4.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uID</th>\n",
       "      <th>RBP</th>\n",
       "      <th>Cell line</th>\n",
       "      <th>CLIP_rep1</th>\n",
       "      <th>CLIP_rep2</th>\n",
       "      <th>INPUT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>203</td>\n",
       "      <td>HNRNPC</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>204</td>\n",
       "      <td>RBFOX2</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>205</td>\n",
       "      <td>IGF2BP1</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>206</td>\n",
       "      <td>HNRNPK</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>209</td>\n",
       "      <td>SRSF7</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   uID      RBP Cell line                                          CLIP_rep1  \\\n",
       "0  203   HNRNPC     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "1  204   RBFOX2     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "2  205  IGF2BP1     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "3  206   HNRNPK     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "4  209    SRSF7     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "\n",
       "                                           CLIP_rep2  \\\n",
       "0  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "1  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "2  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "3  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "4  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "\n",
       "                                               INPUT  \n",
       "0  /projects/ps-yeolab3/encode/analysis/encode_GR...  \n",
       "1  /projects/ps-yeolab3/encode/analysis/encode_GR...  \n",
       "2  /projects/ps-yeolab3/encode/analysis/encode_GR...  \n",
       "3  /projects/ps-yeolab3/encode/analysis/encode_GR...  \n",
       "4  /projects/ps-yeolab3/encode/analysis/encode_GR...  "
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "encode3.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>uid</th>\n",
       "      <th>Batch</th>\n",
       "      <th>RBP</th>\n",
       "      <th>prefix</th>\n",
       "      <th>bam_0</th>\n",
       "      <th>bam_1</th>\n",
       "      <th>bam_control</th>\n",
       "      <th>plus_0</th>\n",
       "      <th>plus_1</th>\n",
       "      <th>plus_control</th>\n",
       "      <th>minus_0</th>\n",
       "      <th>minus_1</th>\n",
       "      <th>minus_control</th>\n",
       "      <th>bed_0</th>\n",
       "      <th>bed_1</th>\n",
       "      <th>idr</th>\n",
       "      <th>Cell Line</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>17</td>\n",
       "      <td>4037</td>\n",
       "      <td>batch11a</td>\n",
       "      <td>EEF2</td>\n",
       "      <td>encode4_batch11a.4037</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>K562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>22</td>\n",
       "      <td>4050</td>\n",
       "      <td>batch11b</td>\n",
       "      <td>DDX47</td>\n",
       "      <td>encode4_batch11b.4050</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>K562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>29</td>\n",
       "      <td>4048</td>\n",
       "      <td>batch12</td>\n",
       "      <td>SF3A1</td>\n",
       "      <td>encode4_batch12.4048</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>K562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>43</td>\n",
       "      <td>4107</td>\n",
       "      <td>batch17</td>\n",
       "      <td>EEF2</td>\n",
       "      <td>encode4_batch17.4107</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>HepG2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>48</td>\n",
       "      <td>4117</td>\n",
       "      <td>batch17</td>\n",
       "      <td>SF3A1</td>\n",
       "      <td>encode4_batch17.4117</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>/projects/ps-yeolab5/encore/processing/encore_...</td>\n",
       "      <td>HepG2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Unnamed: 0   uid     Batch    RBP                 prefix  \\\n",
       "17          17  4037  batch11a   EEF2  encode4_batch11a.4037   \n",
       "22          22  4050  batch11b  DDX47  encode4_batch11b.4050   \n",
       "29          29  4048   batch12  SF3A1   encode4_batch12.4048   \n",
       "43          43  4107   batch17   EEF2   encode4_batch17.4107   \n",
       "48          48  4117   batch17  SF3A1   encode4_batch17.4117   \n",
       "\n",
       "                                                bam_0  \\\n",
       "17  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "22  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "29  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "43  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "48  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "\n",
       "                                                bam_1  \\\n",
       "17  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "22  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "29  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "43  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "48  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "\n",
       "                                          bam_control  \\\n",
       "17  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "22  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "29  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "43  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "48  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "\n",
       "                                               plus_0  \\\n",
       "17  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "22  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "29  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "43  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "48  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "\n",
       "                                               plus_1  \\\n",
       "17  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "22  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "29  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "43  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "48  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "\n",
       "                                         plus_control  \\\n",
       "17  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "22  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "29  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "43  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "48  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "\n",
       "                                              minus_0  \\\n",
       "17  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "22  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "29  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "43  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "48  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "\n",
       "                                              minus_1  \\\n",
       "17  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "22  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "29  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "43  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "48  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "\n",
       "                                        minus_control  \\\n",
       "17  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "22  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "29  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "43  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "48  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "\n",
       "                                                bed_0  \\\n",
       "17  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "22  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "29  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "43  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "48  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "\n",
       "                                                bed_1  \\\n",
       "17  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "22  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "29  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "43  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "48  /projects/ps-yeolab5/encore/processing/encore_...   \n",
       "\n",
       "                                                  idr Cell Line  \n",
       "17  /projects/ps-yeolab5/encore/processing/encore_...      K562  \n",
       "22  /projects/ps-yeolab5/encore/processing/encore_...      K562  \n",
       "29  /projects/ps-yeolab5/encore/processing/encore_...      K562  \n",
       "43  /projects/ps-yeolab5/encore/processing/encore_...     HepG2  \n",
       "48  /projects/ps-yeolab5/encore/processing/encore_...     HepG2  "
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "encode4.loc[encode4['RBP'].isin(essential_gene)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uID</th>\n",
       "      <th>RBP</th>\n",
       "      <th>Cell line</th>\n",
       "      <th>CLIP_rep1</th>\n",
       "      <th>CLIP_rep2</th>\n",
       "      <th>INPUT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>203</td>\n",
       "      <td>HNRNPC</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>228</td>\n",
       "      <td>SF3B4</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>242</td>\n",
       "      <td>U2AF2</td>\n",
       "      <td>K562</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>244</td>\n",
       "      <td>U2AF1</td>\n",
       "      <td>K562</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>249</td>\n",
       "      <td>HNRNPU</td>\n",
       "      <td>K562</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>272</td>\n",
       "      <td>U2AF2</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>281</td>\n",
       "      <td>HNRNPU</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>282</td>\n",
       "      <td>U2AF1</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>311</td>\n",
       "      <td>XRCC6</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>331</td>\n",
       "      <td>RBM22</td>\n",
       "      <td>K562</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>480</td>\n",
       "      <td>XRCC6</td>\n",
       "      <td>K562</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>494</td>\n",
       "      <td>RBM22</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>506</td>\n",
       "      <td>SF3B1</td>\n",
       "      <td>K562</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>507</td>\n",
       "      <td>SF3B4</td>\n",
       "      <td>K562</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125</th>\n",
       "      <td>545</td>\n",
       "      <td>EIF3H</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>548</td>\n",
       "      <td>ZRANB2</td>\n",
       "      <td>K562</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>556</td>\n",
       "      <td>CDC40</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165</th>\n",
       "      <td>646</td>\n",
       "      <td>EIF3G</td>\n",
       "      <td>K562</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176</th>\n",
       "      <td>677</td>\n",
       "      <td>GEMIN5</td>\n",
       "      <td>K562</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>685</td>\n",
       "      <td>WDR3</td>\n",
       "      <td>K562</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193</th>\n",
       "      <td>699</td>\n",
       "      <td>HNRNPC</td>\n",
       "      <td>K562</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>215</th>\n",
       "      <td>752</td>\n",
       "      <td>PRPF4</td>\n",
       "      <td>HepG2</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "      <td>/projects/ps-yeolab3/encode/analysis/encode_GR...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     uID     RBP Cell line                                          CLIP_rep1  \\\n",
       "0    203  HNRNPC     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "15   228   SF3B4     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "21   242   U2AF2      K562  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "23   244   U2AF1      K562  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "26   249  HNRNPU      K562  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "32   272   U2AF2     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "36   281  HNRNPU     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "37   282   U2AF1     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "47   311   XRCC6     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "54   331   RBM22      K562  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "98   480   XRCC6      K562  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "105  494   RBM22     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "110  506   SF3B1      K562  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "111  507   SF3B4      K562  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "125  545   EIF3H     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "127  548  ZRANB2      K562  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "132  556   CDC40     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "165  646   EIF3G      K562  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "176  677  GEMIN5      K562  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "183  685    WDR3      K562  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "193  699  HNRNPC      K562  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "215  752   PRPF4     HepG2  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "\n",
       "                                             CLIP_rep2  \\\n",
       "0    /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "15   /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "21   /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "23   /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "26   /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "32   /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "36   /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "37   /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "47   /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "54   /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "98   /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "105  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "110  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "111  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "125  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "127  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "132  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "165  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "176  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "183  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "193  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "215  /projects/ps-yeolab3/encode/analysis/encode_GR...   \n",
       "\n",
       "                                                 INPUT  \n",
       "0    /projects/ps-yeolab3/encode/analysis/encode_GR...  \n",
       "15   /projects/ps-yeolab3/encode/analysis/encode_GR...  \n",
       "21   /projects/ps-yeolab3/encode/analysis/encode_GR...  \n",
       "23   /projects/ps-yeolab3/encode/analysis/encode_GR...  \n",
       "26   /projects/ps-yeolab3/encode/analysis/encode_GR...  \n",
       "32   /projects/ps-yeolab3/encode/analysis/encode_GR...  \n",
       "36   /projects/ps-yeolab3/encode/analysis/encode_GR...  \n",
       "37   /projects/ps-yeolab3/encode/analysis/encode_GR...  \n",
       "47   /projects/ps-yeolab3/encode/analysis/encode_GR...  \n",
       "54   /projects/ps-yeolab3/encode/analysis/encode_GR...  \n",
       "98   /projects/ps-yeolab3/encode/analysis/encode_GR...  \n",
       "105  /projects/ps-yeolab3/encode/analysis/encode_GR...  \n",
       "110  /projects/ps-yeolab3/encode/analysis/encode_GR...  \n",
       "111  /projects/ps-yeolab3/encode/analysis/encode_GR...  \n",
       "125  /projects/ps-yeolab3/encode/analysis/encode_GR...  \n",
       "127  /projects/ps-yeolab3/encode/analysis/encode_GR...  \n",
       "132  /projects/ps-yeolab3/encode/analysis/encode_GR...  \n",
       "165  /projects/ps-yeolab3/encode/analysis/encode_GR...  \n",
       "176  /projects/ps-yeolab3/encode/analysis/encode_GR...  \n",
       "183  /projects/ps-yeolab3/encode/analysis/encode_GR...  \n",
       "193  /projects/ps-yeolab3/encode/analysis/encode_GR...  \n",
       "215  /projects/ps-yeolab3/encode/analysis/encode_GR...  "
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "encode3.loc[encode3['RBP'].isin(essential_gene)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "# we found the essential genes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Fisher Exact Test demo using RBM22"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [],
   "source": [
    "# What are the components of this command?\n",
    "uid = encode3.loc[(encode3['RBP']=='RBM22')&(encode3['Cell line']=='HepG2'), 'uID'].iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'494'"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [],
   "source": [
    "idr_path = '/home/hsher/seqdata/eclip_bed/sorted/{}.01v02.IDR.out.0102merged.bed.blacklist_removed.bed.narrowPeak.bed'.format(uid)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/hsher/seqdata/eclip_bed/sorted/494.01v02.IDR.out.0102merged.bed.blacklist_removed.bed.narrowPeak.bed'"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "idr_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [],
   "source": [
    "idr = BedTool(idr_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<BedTool(/home/hsher/seqdata/eclip_bed/sorted/494.01v02.IDR.out.0102merged.bed.blacklist_removed.bed.narrowPeak.bed)>"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "idr"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Goal:\n",
    "- does our RBP (RBM22) regulate tumor-essential genes more than other genes? (likelihood)\n",
    "\n",
    "4 groups we need to find:\n",
    "1. tumor essential AND bound by RBM22\n",
    "2. tumor essential and NOT bound by RBM22\n",
    "3. NOT tumor essential AND bound by RBM22\n",
    "4. NOT tumor essential and NOT bound by RBM22\n",
    "\n",
    "Use BedTool intersect to find the 4 groups:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [],
   "source": [
    "# group 1: essential genes AND bound by RBM22\n",
    "group_1 = essential_gene_coords.intersect(idr, s = True, u = True).saveas()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [],
   "source": [
    "# group 2: essential genes and NOT bound by RBM22\n",
    "group_2 = essential_gene_coords.intersect(idr, s = True, v = True).saveas()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [],
   "source": [
    "# transcripts that are not tumor essential genes, so we can repeat this process for group 3 & 4\n",
    "nonessential_gene_coords = transcript_coords.intersect(essential_gene_coords, s = True, v = True).saveas()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [],
   "source": [
    "# group 3: nonessential genes AND bound by RBM22\n",
    "group_3 = nonessential_gene_coords.intersect(idr, s = True, u = True).saveas()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "# group 4: nonessential genes and NOT bound by RBM22\n",
    "group_4 = nonessential_gene_coords.intersect(idr, s = True, v = True).saveas()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [],
   "source": [
    "# counting genes for table\n",
    "table_for_stat = pd.DataFrame([\n",
    "    [len(group_1), len(group_2)], \n",
    "     [len(group_3), len(group_4)]\n",
    "], \n",
    "    \n",
    "    columns = ['bound by RBM22', 'not bound'],\n",
    "    index = ['essential', 'not essential']\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bound by RBM22</th>\n",
       "      <th>not bound</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>essential</th>\n",
       "      <td>26</td>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>not essential</th>\n",
       "      <td>1244</td>\n",
       "      <td>58719</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               bound by RBM22  not bound\n",
       "essential                  26        102\n",
       "not essential            1244      58719"
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table_for_stat\n",
    "# things to note - syntax for putting in rows [square brackets]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notes from Charlene on stats\n",
    "- scipy stats documentation: https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.fisher_exact.html\n",
    "- odds ratio from NIH: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2938757/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "from scipy.stats import fisher_exact"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [],
   "source": [
    "odds, p = fisher_exact(table_for_stat, alternative = 'two-sided')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "12.031823340268584"
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "odds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "12.031823340268582"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# how to calculate odds ratio\n",
    "(26/1244)/(102/58719)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2.793388556985412e-18"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [],
   "source": [
    "# very low p value means it's significant! :D"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Conclusion: we can conclude that tumor essential genes are regulated by RBM22, meaning they have IDR peaks, more than expected"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
